connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -2326,10 +2328,21 @@ class SparkConnectPlanner(val session: SparkSession) {
 
       case StreamingQueryCommand.CommandCase.EXCEPTION =>
         val result = query.exception
-        result.foreach(e =>
-          respBuilder.getExceptionBuilder
+        if (result.isDefined) {
+          val e = result.get
+          val exception_builder = StreamingQueryCommandResult.ExceptionResult
+            .newBuilder()
+          exception_builder
             .setExceptionMessage(SparkConnectService.abbreviateErrorMessage(e.toString))
-            .setErrorClass(e.getErrorClass))
+            .setErrorClass(e.getErrorClass)
+
+          val maxSize = SparkEnv.get.conf.get(CONNECT_JVM_STACK_TRACE_MAX_SIZE)
+          val stackTrace = Option(ExceptionUtils.getStackTrace(e))
+          stackTrace.foreach { s =>
+            exception_builder.setStackTrace(StringUtils.abbreviate(s, maxSize))
```

Comment:
The error message and stack trace are truncated due to the HTTP header size limitation (default 8k). I noticed that you're not throwing this as a StatusRuntimeException. If the same limitation doesn't apply here, you could use untruncated messages and stack traces.

Reply:
Thanks! That'd be great! This exception() method was meant to provide more information including the logical plan. If it can be done without truncation that's even better

Reply:
+1 it seems that you are using proto itself to carry the exception information thus the limitation does not apply.